### ISSUE TITLE
Create organization repo and transfer Auto-GPT there to allow granular roles & permissions for collaborators

### ISSUE THREAD

**Issue by JuroOravec** [Significant-Gravitas/Auto-GPT#1067]
### Duplicates

- [X] I have searched the existing issues

### Summary 💡

**TL;DR: Create org and transfer the repo there.**

Issue:

Current repo setup (private repo) makes it hard for volunteers to help with issue/project maintenance.

As @richbeales said in [#609](https://github.com/Torantulino/Auto-GPT/issues/609):
> I discussed this earlier with @Torantulino currently the github repo is a personal one, so there's no way of doing granular permissions without converting to an organisation account.
Currently contributor role is all-or-nothing regarding write access, which he is (rightly) quite nervous about giving out to randomers on the internet.

I came upon this when I wanted to go over the currently open issues and PRs and label them / clean up (But obvs can't because I don't have permissions).

Given the popularity of the repo, I think that moving it under an org is a good next step. Eg for collaborators who want to help with maintaining issues / discussions / PRs, there's a [Triage role](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization).

Steps for @Torantulino:

1) [Create organization repo](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/creating-a-new-organization-from-scratch)
   - Read [More about organizations on GitHub](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/about-organizations)
 
2) [Transfer Auto-GPT to the organization](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository)
   - Is it safe to transfer the repo? Mostly yes, most things are automatically moved or redirected. It looks like only sponsors might be affected. [More about transfer here](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository#whats-transferred-with-a-repository)
      > Sponsors who have access to the repository through a sponsorship tier may be affected. For more information, see "[Managing your sponsorship tiers](https://docs.github.com/en/sponsors/receiving-sponsorships-through-github-sponsors/managing-your-sponsorship-tiers#adding-a-repository-to-a-sponsorship-tier)".

3) [Configure org permissions for active collaborators](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/setting-base-permissions-for-an-organization) (default permission for members is READ)
   - [See here for the breakdown of all roles](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization)

4) ???

5) Profit

---

Note: There's also the option to [convert personal account to organization](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-personal-account/converting-a-user-into-an-organization). I don't recommending, but mentioning it for awareness.
